specs/appwindow-spec.md

```diff
@@ -0,0 +1,104 @@
+functionality through WinUI 3 code
+
+# Background
+Xaml has a [Window](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Window) API that
+internally wraps an hwnd. Windows has an [AppWindow](https://docs.microsoft.com/uwp/api/Windows.UI.WindowManagement.AppWindow) class
+that similarly wraps an hwnd in UWP. WinAppSDK has a new [AppWindow](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/Microsoft.UI.Windowing.AppWindow) which wraps an hwnd and works on Desktop.
+
+You can get an AppWindow from a Xaml Window by calling a COM API to get Xaml and then
+a DLL export API to convert the hwnd to an AppWindow. This spec adds a simple `Window.AppWindow`
+property to make this much easier and more discoverable.
+
+[Xaml Window](https://docs.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.window) will
+expose AppWindow object directly to app developer through an api. Instead of writing a lot of boiler plate
+code everywhere, app developer can use this api, reducing code bloat, and making appwindow apis easily
+accessible from WinUI 3 code.
+
+These *before* and *after* C# code examples illustrate how this api simplifies integrating appwindow apis in WinUI 3 codebase.
+
+### Before
+```c#
+// This is needed to get any Window from inside a Xaml control 
+var xamlWindow = WindowHelper.GetWindowForElement(this); // api to get window object from UIElement (not a real api)  
+
+// unnecessary boiler plate code 
+var windowId = Win32Interop.GetWindowIdFromWindow(WindowNative.GetWindowHandle(xamlWindow)); 
+var appWindow = AppWindow.GetFromWindowId(windowId); 
+
+//calling function foo  (not a real appwindow function)
+appWindow.foo();
+```
+
+### After
+```c#
+// This is needed to get any Window from inside a Xaml control 
+var xamlWindow = WindowHelper.GetWindowForElement(this); // api to get window object from UIElement (not a real api)  
+   
+
+//calling function foo directly
+xamlWindow.AppWindow.foo();  
+```
+Notice how `xamlWindow.AppWindow.foo()` doesn't require additional steps to call the function `foo`.
+
+
+# API Pages
+
+## Window.AppWindow property
+
+Gets the `AppWindow` associated with this `Window`.
+```c#
+public Microsoft.UI.Windowing.AppWindow Window.AppWindow { get; }
+```
+### Sample example
+```c#
+var xamlWindow = WindowHelper.GetWindowForElement(this);   
+auto windowSize = xamlWindow.AppWindow.Size;  
+```
+
+Mapping an hwnd to an AppWindow causes the hwnd to be subclassed, meaning that the timing of
+when [AppWindow.GetFromWindowId](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/Microsoft.UI.Windowing.AppWindow.GetFromWindowId)
+is first called can potentially have a behavioral side effect. To ensure this is predictable, the AppWindow
+for Xaml's hwnd will be created during the Xaml Window's construction. Note that this could potentially be
+an observable behavior change from the behavior before the introduction of this API.
+
+New subclassing order with this new feature change:
```

Comment:
If I derive from Window and subclass myself as well would the position of the subclass change in the order?

Reply:
When you say "subclass" you're talking about the hwnd subclass (SetWindowSubclass)? You can change the order if you create the AppWindow ahead of the Window, whether deriving or not.

Reply:
With subclass I mean SetWindowLongPtr(hwnd, GWL_WNDPROC, newWndProc)

Reply:
The first time an AppWindow is created for an hwnd (`AppWindow.GetFromWindowId`) it will add its subclass. So any subclassing you do before/after that will be in that order. And also if you create the AppWindow before creating the Xaml Window, you'll cause it to be potentially earlier in the order.